test/e2e-go/restAPI/restClient_test.go

```diff
@@ -1478,10 +1473,22 @@ end:
 		`∑´´˙©˚¬∆ßåƒ√¬`,
 	}
 
+	// Happy paths:
 	resp, err := testClient.ApplicationBoxes(uint64(createdAppID), 0)
 	a.NoError(err)
 	a.Empty(resp.Boxes)
 
+	// Some Un-Happy paths:
+
+	nonexistantAppIndex := uint64(1337)
+	_, err = testClient.ApplicationInformation(nonexistantAppIndex)
+	a.ErrorContains(err, "application does not exist")
+	_, err = testClient.ApplicationBoxes(nonexistantAppIndex, 0)
+	// TODO: Issue #4575 asserts that this SHOULD be erroring
+	a.NoError(err)
```

Comment:
In my opinion, this _ought to be returning a 404 Not Found_, but instead it's not erroring at all. I've created issue #4575 to address at a later stage.

Reply:
resolving after conversation with @jannotti on the now closed issue